clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -178,9 +217,15 @@ public UpdateMetadataRequest(Struct struct, short versionId) {
             for (Object r : replicasArray)
                 replicas.add((Integer) r);
 
-            PartitionState partitionState = new PartitionState(controllerEpoch, leader, leaderEpoch, isr, zkVersion, replicas);
-            partitionStates.put(new TopicPartition(topic, partition), partitionState);
+            Object[] offlineReplicasArray = partitionStateData.hasField(OFFLINE_REPLICAS_KEY_NAME) ?
+                partitionStateData.getArray(OFFLINE_REPLICAS_KEY_NAME) : new Object[0];
+            List<Integer> offlineReplicas = new ArrayList<>(offlineReplicasArray.length);
```

Comment:
The current code seems simpler.. I think the extra `if` statement has negligible impact on performance, right?